packages/controller/lib/setup/setupRepo.js

```diff
@@ -227,154 +215,115 @@ function Repo(options) {
-                            let activeRepo = obj.common.activeRepo;
-                            if (typeof activeRepo === 'string') {
-                                activeRepo = [activeRepo];
-                            }
-                            console.log(`\nActive repo(s): ${activeRepo.join(', ')}`);
-                        }
-                        callback();
-                    });
-                } else {
-                    console.error('Cannot get list: ' + err);
-                    callback(EXIT_CODES.CANNOT_GET_REPO_LIST);
+    this.showRepoStatus = async function () {
+        try {
+            const obj = await objects.getObjectAsync('system.repositories');
+            if (!obj) {
+                console.error('List is empty');
+                return EXIT_CODES.CANNOT_GET_REPO_LIST;
+            } else
+            if (obj.native.repositories) {
+                Object.keys(obj.native.repositories).forEach(r =>
+                    console.log(`${r.padEnd(14)}: ${obj.native.repositories[r].link}`));
+
+                const objCfg = await objects.getObjectAsync('system.config');
+                if (objCfg && objCfg.common) {
+                    let activeRepo = objCfg.common.activeRepo;
+                    if (typeof activeRepo === 'string') {
+                        activeRepo = [activeRepo];
+                    }
+                    console.log(`\nActive repo(s): ${activeRepo.join(', ')}`);
                 }
+            } else {
+                console.error('List is empty');
+                return EXIT_CODES.CANNOT_GET_REPO_LIST;
             }
-        });
+        } catch (err)  {
+            console.error('Cannot get list: ' + err);
+            return EXIT_CODES.CANNOT_GET_REPO_LIST;
+        }
     };
 
-    this.add = function (repoName, repoUrl, callback) {
-        objects.getObject('system.repositories', (err, obj) => {
-            if (err) {
-                callback && callback(err);
-            } else
-            if (!obj) {
-                obj = defaultSystemRepo;
-            }
+    this.add = async function (repoName, repoUrl) {
+        let obj = await objects.getObjectAsync('system.repositories');
+        obj = obj || defaultSystemRepo;
 
-            if (obj.native.repositories[repoName]) {
-                callback && callback(`Repository "${repoName}" yet exists: ${obj.native.repositories[repoName].link}`);
-            } else {
-                obj.native.repositories[repoName] = {
-                    link: repoUrl,
-                    json: null
-                };
-                obj.from = 'system.host.' + tools.getHostName() + '.cli';
-                obj.ts = Date.now();
-                objects.setObject('system.repositories', obj, callback);
-            }
-        });
+        if (obj.native.repositories[repoName]) {
+            return `Repository "${repoName}" yet exists: ${obj.native.repositories[repoName].link}`;
```

Comment:
prevously this was first cb arg so probably an error argument, should it not be a throw here then?